docker/docker.go

```diff
@@ -24,11 +24,118 @@ const (
 	defaultCertFile     = "cert.pem"
 )
 
+func checkDaemon(args []string) bool {
+	for j := 1; j < len(args); j++ {
+		if args[j] == "-d" || args[j] == "--daemon" {
+			return true
+		} else if args[j][:1] == "-" {
+			if index_of_equal := strings.Index(args[j], "="); index_of_equal != -1 {
+				continue //Flag with a value
+			} else {
+				isBool, exists := flag.CommandLine.FlagsMap[args[j]]
+				if exists && !isBool {
+					//Global string flag
+					j = j + 1
+					continue
+				} else if exists && isBool {
+					//Global bool flag
+					continue
+				}
+
+				isBool, exists = cmd.FlagsMap[args[j]]
+				if exists && !isBool {
+					//Daemon string flag
+					j = j + 1
+					continue
+				} else if exists && isBool {
+					//Daemon bool flag
+					continue
+				}
+			}
+		} else {
+			for _, command := range Commands {
+				if args[j] == command[0] {
+					return false
+				}
+			}
```

Comment:
Why not just return false here instead of this loop?
If it's not a command it's probably a misspelling of a command, so there's no point in shuffling.